internal/web_server/web/handlers/api/admin_api.go

```diff
@@ -310,25 +318,38 @@ func (aah *AdminApiHandler) createQuestion(c echo.Context) error {
 		return echo.NewHTTPError(http.StatusBadRequest, errorText)
 	}
 
-	// Check if the article URL is in the database
-	article, err := articles.GetArticleByURL(aah.sharedData.DB, articleURL)
-	if err != nil && err != sql.ErrNoRows {
-		return echo.NewHTTPError(http.StatusBadRequest, "Failed to get article ID")
-	}
+	article := &articles.Article{}
 
-	// If not in DB, fetch the relevant article data and add it to the DB
-	if article == nil {
-		tempArticle, err := articles.GetSmpArticleByURL(articleURLString)
-		if err != nil {
-			return echo.NewHTTPError(http.StatusBadRequest, "Failed to fetch article data")
+	// Only add article to DB if it is not empty.
+	// I.e. allow for no article, but not invalid article.
+	if articleURLString != "" {
+		// Check if the article URL is in the database
+		article, err = articles.GetArticleByURL(aah.sharedData.DB, articleURL)
+		if err != nil && err != sql.ErrNoRows {
+			return echo.NewHTTPError(http.StatusBadRequest, "Failed to get article ID")
 		}
 
-		articles.AddArticle(aah.sharedData.DB, &tempArticle)
-		article = &tempArticle
+		// If not in DB, fetch the relevant article data and add it to the DB
+		if article == nil {
+			tempArticle, err := articles.GetSmpArticleByURL(articleURLString)
+			if err != nil {
+				return echo.NewHTTPError(http.StatusBadRequest, "Failed to fetch article data")
```

Comment:
is tat a badrequest? is client at fault?

Reply:
Unsure. It is their responsibility to check that the article URL is valid. But being unable to fetch data could also be due to server error. The only way to check if the URL is valid is to attempt fetching the data though.

Reply:
I guess I can check what error `GetSmpArticleByURL` returns. Will make my own error types.